### PR TITLE
Fixes and mining stuff

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

This pull request fixes xenomorphs being able to use complex items through /give(), while adding some features and balancing existing ones.

Megafauna is able to gib miners with Memento Mori once again.
Goliath cloak was severely nerfed, however, now they can be equipped in a cloak slot instead of suit one. You still can equip it's hood.
New hardsuit was added! It's called "Protosuit" and it is a failed NT experiment in recreating Syndicate hardsuits for medical purposes. It has a medical heads-up display and will recognise user's name from DNA samples, only working on users who has medical records. It can be found in tendril chests

## Why It's Good For The Game

Bugfixes and balance are always great. Also, mining only had 1 normal hardsuit and other one was shitty.

## Changelog
:cl:
add: New "Protosuit" hardsuit was added!
tweak: Goliath cloak is now a propper cloak
fix: Megafauna can now once again gib miners with Memento Mori
fix: Xenomorphs can no longer pick up human items
/:cl:

### CODE REVIEW

**Comment by Whoneedspacee** [comment]
> PKAs now only gain 0.75 of their existing cooldown for every other existing PKA in player's inventory. Just because they really needed a buff

what, why do they need a buff, this is to prevent gimmicky shit like stacking 10 pkas in your inventory and this just enables that to be a straight up buff for dual wielding pkas not even considering swapping out spares from your backpack

the pka is and always will be the best weapon for rushing roundstart, let crusher be the scaled weapon that does well with a hierophant upgrade you have to grab every single round

everything else cool

File: code/modules/mob/living/carbon/inventory.dm [hunk 1]
```diff
@@ -172,7 +172,7 @@
 		return
 	visible_message("<span class='notice'>[src] is offering [receiving]</span>", \
 					"<span class='notice'>You offer [receiving]</span>", null, 2)
-	for(var/mob/living/carbon/C in orange(1, src))
+	for(var/mob/living/carbon/human/C in orange(1, src)) //Fixed that, now it shouldn't be able to give benos stunbatons and IDs
```
Comment (Fikou) [inline code/modules/mob/living/carbon/inventory.dm:175]: atomize?

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -300,6 +300,61 @@
 	. = ..()
 	AddComponent(/datum/component/armor_plate)
 
+
+//Nanotrasen medical Protosuit
+
+/obj/item/clothing/suit/space/hardsuit/protosuit
+	name = "protosuit hardsuit"
+	desc = "A failed NanoTrasen experiment, intended to be a unique medical hardsuit based on stolen Syndicate samples."
```
Comment (Fikou) [inline code/modules/clothing/spacesuits/hardsuit.dm:308]: not spelled NanoTrasen
also this is a magic tendril item, why nanotrasen lore?

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 3]
```diff
@@ -300,6 +300,61 @@
...
+	else
+		if(initialized)
+			REMOVE_TRAIT(user, TRAIT_SELF_AWARE, CLOTHING_TRAIT)
+			REMOVE_TRAIT(user, TRAIT_MEDICAL_HUD, CLOTHING_TRAIT)
+			display_visor_message("<span class='notice'>De-initializing HUD system. Have a great shift.</span>")
+			initialized = FALSE
+
+/obj/item/clothing/head/helmet/space/hardsuit/protosuit/dropped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(initialized)
```
Comment (Fikou) [inline code/modules/clothing/spacesuits/hardsuit.dm:352]: why is this var needed

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 4]
```diff
@@ -300,6 +300,61 @@
...
+				return
+
+			var/datum/data/record/R = find_record("name", user.dna.real_name, GLOB.data_core.medical) //The helmet is based on DNA
+			if(!R)
+				display_visor_message("<span class='notice'>Initialization error. Unable to recognise user. Halting setup process.</span>")
+				return
+
+			display_visor_message("<span class='notice'>Initialization complete. User recognised. Hello, [user.dna.real_name].</span>")
+			ADD_TRAIT(user, TRAIT_SELF_AWARE, CLOTHING_TRAIT)
+			ADD_TRAIT(user, TRAIT_MEDICAL_HUD, CLOTHING_TRAIT)
```
Comment (Fikou) [inline code/modules/clothing/spacesuits/hardsuit.dm:341]: is this supposed to not give you the actual medhud?

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 5]
```diff
@@ -300,6 +300,61 @@
...
+			initialized = TRUE
+	else
+		if(initialized)
+			REMOVE_TRAIT(user, TRAIT_SELF_AWARE, CLOTHING_TRAIT)
+			REMOVE_TRAIT(user, TRAIT_MEDICAL_HUD, CLOTHING_TRAIT)
+			display_visor_message("<span class='notice'>De-initializing HUD system. Have a great shift.</span>")
+			initialized = FALSE
+
+/obj/item/clothing/head/helmet/space/hardsuit/protosuit/dropped(mob/living/carbon/human/user, slot)
```
Comment (Fikou) [inline code/modules/clothing/spacesuits/hardsuit.dm:350]: dropped doesnt have a slot argument

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 6]
```diff
@@ -300,6 +300,61 @@
...
+
+//Nanotrasen medical Protosuit
+
+/obj/item/clothing/suit/space/hardsuit/protosuit
+	name = "protosuit hardsuit"
+	desc = "A failed NanoTrasen experiment, intended to be a unique medical hardsuit based on stolen Syndicate samples."
+	icon_state = "protosuit"
+	hardsuit_type = "protosuit"
+	armor = list(MELEE = 30, BULLET = 15, LASER = 15, ENERGY = 25, BOMB = 30, BIO = 100, RAD = 25, FIRE = 75, ACID = 75, WOUND = 15)
+	allowed = allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/storage/firstaid, /obj/item/healthanalyzer, /obj/item/stack/medical)
```
Comment (Fikou) [inline code/modules/clothing/spacesuits/hardsuit.dm:312]:
```suggestion
	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/storage/firstaid, /obj/item/healthanalyzer, /obj/item/stack/medical)
```

Comment (Fikou) [reply code/modules/clothing/spacesuits/hardsuit.dm:312]: this doesnt compile, did you test the pr

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 5 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/hardsuit.dm:350]:
```suggestion
/obj/item/clothing/head/helmet/space/hardsuit/protosuit/dropped()
```
you dont use user either

Comment (Fikou) [reply code/modules/clothing/spacesuits/hardsuit.dm:350]:
```suggestion
/obj/item/clothing/head/helmet/space/hardsuit/protosuit/dropped(mob/living/carbon/human/user)
```

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (SmArtKar) [reply code/modules/clothing/spacesuits/hardsuit.dm:308]: same reason as NT brand disks

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 7]
```diff
@@ -300,6 +300,61 @@
 	. = ..()
 	AddComponent(/datum/component/armor_plate)
 
+
+//Nanotrasen medical Protosuit
+
+/obj/item/clothing/suit/space/hardsuit/protosuit
+	name = "protosuit hardsuit"
+	desc = "A failed NanoTrasen experiment, intended to be a unique medical hardsuit based on stolen Syndicate samples."
+	icon_state = "protosuit"
```
Comment (Fikou) [inline code/modules/clothing/spacesuits/hardsuit.dm:309]: you didnt add icons for these

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (SmArtKar) [reply code/modules/clothing/spacesuits/hardsuit.dm:308]: Also it IS spelled NanoTrasen, just searched in-code

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 3 above)
Comment (SmArtKar) [reply code/modules/clothing/spacesuits/hardsuit.dm:352]: To know should you display the exit message or not

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 8]
```diff
@@ -122,6 +122,10 @@
 				OpenFire()
 		else
 			devour(L)
+			return
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:125]: dont do this

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 7 above)
Comment (SmArtKar) [reply code/modules/clothing/spacesuits/hardsuit.dm:309]: oh fuck forgot to put em in

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 9]
```diff
@@ -122,6 +122,10 @@
...
+
+		if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still gibs yall
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:127]: turn this into an else if between the if and else

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 8 above)
Comment (SmArtKar) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:125]: yeah sure

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/hardsuit.dm:308]: no its Nanotrasen

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 3 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/hardsuit.dm:352]: it should always display it?

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 9 above)
Comment (SmArtKar) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:127]: yeah

**Comment by GuillaumePrata** [comment]
Giving Miners a extra 10 melee brings brings it closer and closer to 100 (Which I have no idea if it's actually a big deal but...)
Ash Drake is 70, Sec Jumpsuit 10, Bone Talisman 5, Cloak 10 = 95  Melee armor.

Another thing that might not be important, but the Goliath Cloak was kinda useful if you wanted to throw spears at miners as a Ashwalker since it's the only armor (I think) that can carry spears on it's exoslot.

Still if going this way, wouldn't it be better to make the cloak storm proof so miners could stay in the storm with it instead of getting extra armor? Would make it valuable for other Lavaland spawners too since they can't drop a pod if they are too far from safety and a lot more interesting gameplay wise than extra armor.

**Comment by SmArtKar** [comment]
> Giving Miners a extra 10 melee brings brings it closer and closer to 100 (Which I have no idea if it's actually a big deal but...)
> Ash Drake is 70, Sec Jumpsuit 10, Bone Talisman 5, Cloak 10 = 95 Melee armor.
> 
> Another thing that might not be important, but the Goliath Cloak was kinda useful if you wanted to throw spears at miners as a Ashwalker since it's the only armor (I think) that can carry spears on it's exoslot.
> 
> Still if going this way, wouldn't it be better to make the cloak storm proof so miners could stay in the storm with it instead of getting extra armor? Would make it valuable for other Lavaland spawners too since they can't drop a pod if they are too far from safety and a lot more interesting gameplay wise than extra armor.

Here's the deal, it was 85 and now its 95. Yeah, thats not good, you are right here. But giving players very easy to access stormproof gear is also a really bad idea through

**Comment by Fikou** [comment]
> Here's the deal, it was 85 and now its 95. Yeah, thats not good, you are right here. But giving players very easy to access stormproof gear is also a really bad idea through

is it very easy?
you have to waste time making leather, and kill a bunch of mobs, compared to just killing a tendril and getting a free hardsuit with ash protection, or fighting one boss and getting that protection for 3 different people
why is it a really bad idea?

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 10]
```diff
@@ -127,7 +127,7 @@
 	if(!unique_frequency)
 		for(var/obj/item/gun/energy/kinetic_accelerator/K in loc.GetAllContents())
 			if(!K.unique_frequency)
-				carried++
+				carried += 1
```
Comment (bobbahbrown) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:130]: why

File: code/modules/clothing/suits/toggles.dm [hunk 11]
```diff
@@ -4,6 +4,7 @@
 	actions_types = list(/datum/action/item_action/toggle_hood)
 	var/obj/item/clothing/head/hooded/hood
 	var/hoodtype = /obj/item/clothing/head/hooded/winterhood //so the chaplain hoodie or other hoodies can override this
+	var/toggle_slot = ITEM_SLOT_OCLOTHING
```
Comment (bobbahbrown) [inline code/modules/clothing/suits/toggles.dm:7]: This does not meet our [documentation standards](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md).

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 12]
```diff
@@ -300,6 +300,67 @@
...
+	desc = "A failed Nanotrasen experiment, intended to be a unique medical hardsuit based on stolen Syndicate samples."
+	icon_state = "protosuit"
+	hardsuit_type = "protosuit"
+	armor = list(MELEE = 30, BULLET = 15, LASER = 15, ENERGY = 25, BOMB = 30, BIO = 100, RAD = 25, FIRE = 75, ACID = 75, WOUND = 15)
+	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/storage/firstaid, /obj/item/healthanalyzer, /obj/item/stack/medical)
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/protosuit
+
+/obj/item/clothing/head/helmet/space/hardsuit/protosuit
+	name = "protosuit hardsuit helmet"
+	desc = "A helmet, made in old failed Nanotrasen experiment, which was intended to create a medical hardsuit based of stolen Syndicate ones. You can see a small DNA-sampler on the bottom of it."
```
Comment (bobbahbrown) [inline code/modules/clothing/spacesuits/hardsuit.dm:317]: The wording of this description is awkward and contains a few spelling errors, please change this.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 13]
```diff
@@ -300,6 +300,67 @@
...
+	icon_state = "protosuit"
+	hardsuit_type = "protosuit"
+	armor = list(MELEE = 30, BULLET = 15, LASER = 15, ENERGY = 25, BOMB = 30, BIO = 100, RAD = 25, FIRE = 75, ACID = 75, WOUND = 15)
+	light_range = 7
+	resistance_flags = FIRE_PROOF
+
+	var/initialized = FALSE
```
Comment (bobbahbrown) [inline code/modules/clothing/spacesuits/hardsuit.dm:324]: This does not meet our [documentation standards](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md).

**Comment by bobbahbrown** [comment]
This PR feels as though it could be split up into several more atomic PRs which specifically deal with similar issues, this may negatively impact the review timeline for this PR.

**Comment by IndieanaJones** [comment]
Xeno fix should be split into a separate PR

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 13 above)
Comment (Whoneedspacee) [reply code/modules/clothing/spacesuits/hardsuit.dm:324]: comment yer code

File: code/modules/clothing/suits/toggles.dm (hunk 11 above)
Comment (Whoneedspacee) [reply code/modules/clothing/suits/toggles.dm:7]: doc yer code

**Comment by ArcaneMusic** [comment]
Yeah this needs to be atomised badly: 
Hardsuit on it's own is likely to be denied on par with the fact that we have too many hardsuits as it stands.
Bug fixes are nice, but don't lump them in with unrelated features.
Xeno's having objects in hand is another important fix, but it's unrelated to the megafauna fix as well.

**Comment by SmArtKar** [comment]
Gonna split it

**Pull request closed without merging**